packages/webview_flutter/lib/webview_flutter.dart

```diff
@@ -254,8 +254,22 @@ class WebViewController {
     return _channel.invokeMethod("reload");
   }
 
-  Future<void> _updateSettings(Map<String, dynamic> update) async {
-    return _channel.invokeMethod('updateSettings', update);
+  Future<void> _updateSettings(_WebSettings update) async {
+    Map<String, dynamic> updateMap;
+    if (_settings == null) {
+      updateMap = update.toMap();
+    } else {
+      updateMap = _settings.updatesMap(update);
+    }
+    _settings = update;
+    return _channel.invokeMethod('updateSettings', updateMap);
+  }
+
+  /// Evaluates JavaScript in the context of the current page.
+  /// Returns a Future containing the result of the JavaScript execution.
```

Comment:
What do we know about the Future's type here? it seems that on Android it's always a String (Dartdoc should mention that), what do we know about the type on iOS? (if it's guaranteed to be string on both platforms, we should just make this a `Future<String>` and then we don't need to say anything about the type.)

Reply:
on iOS it returns an ID

Reply:
I meant what do we know about the types of the objects that this id points to?
Is it always a string? is there some mapping from JS types to ObjC classes?
Other than documentation, we should confirm that any object we get as the completionHandler's result can be serialized by the standard message codec and can be sent back over the platform channel.